src/plugins/inappbrowser.ts

```diff
@@ -12,76 +13,72 @@ export interface InAppBrowserEvent extends Event {
+    try {
+      this._objectInstance = cordova.InAppBrowser.open(url, target, options);
+    } catch (e) {
+      window.open(url);
+      console.warn('Native: InAppBrowser is not installed or you are running on a browser. Falling back to window.open, all instance methods will NOT work.');
+    }
+  }
 
   /**
    * Displays an InAppBrowser window that was opened hidden. Calling this has no effect
    * if the InAppBrowser was already visible.
    */
-  show();
+  @CordovaInstance({sync: true})
+  show(): void { }
+
+  /**
+   * Closes the InAppBrowser window.
+   */
+  @CordovaInstance({sync: true})
+  close(): void { }
 
   /**
    * Injects JavaScript code into the InAppBrowser window.
    * @param script    Details of the script to run, specifying either a file or code key.
-   * @param callback  The function that executes after the JavaScript code is injected.
-   *                  If the injected script is of type code, the callback executes with
-   *                  a single parameter, which is the return value of the script, wrapped in an Array.
-   *                  For multi-line scripts, this is the return value of the last statement,
-   *                  or the last expression evaluated.
    */
-  executeScript(script: { file?: string, code?: string }, callback?: (result?: any) => void);
+  @CordovaInstance()
+  executeScript(script: {file?: string, code?: string}): Promise<any> {return; }
 
   /**
    * Injects CSS into the InAppBrowser window.
    * @param css       Details of the script to run, specifying either a file or code key.
-   * @param callback  The function that executes after the CSS is injected.
    */
-  insertCSS(css: { file?: string, code?: string }, callback?: () => void);
-}
-
-@Plugin({
-  plugin: 'cordova-plugin-inappbrowser',
-  pluginRef: 'cordova.InAppBrowser'
-})
-export class InAppBrowser {
+  @CordovaInstance()
+  insertCss(css: {file?: string, code?: string}): Promise<any> {return; }
 
   /**
-   * Opens a URL in a new InAppBrowser instance, the current browser instance, or the system browser.
-   * @param  url     The URL to load.
-   * @param  target  The target in which to load the URL, an optional parameter that defaults to _self.
-   * @param  options Options for the InAppBrowser. Optional, defaulting to: location=yes.
-   *                 The options string must not contain any blank space, and each feature's
-   *                 name/value pairs must be separated by a comma. Feature names are case insensitive.
+   * A method that allows you to listen to events happening in the browser.
+   * @param event Event name
+   * @returns {Observable<any>} Returns back an observable that will listen to the event on subscribe, and will stop listening to the event on unsubscribe.
    */
-  @Cordova({
-    sync: true
-  })
-  static open(url: string, target?: string, options?: string): InAppBrowserRef { return; }
-
+  on(event: string): Observable<InAppBrowserEvent> {
```

Comment:
What's the purpose of this function? Is this generic and something we could abstract for other plugins?